Source/CarthageKit/VersionFile.swift

```diff
@@ -0,0 +1,270 @@
+//
+//  VersionFile.swift
+//  Carthage
+//
+//  Created by Jason Boyle on 8/11/16.
+//  Copyright © 2016 Carthage. All rights reserved.
+//
+
+import Foundation
+import Runes
+import Argo
+import Curry
+import ReactiveSwift
+import ReactiveTask
+import Result
+
+private struct CachedFramework {
+	let name: String
+	let md5: String
+	
+	static let nameKey = "name"
+	static let md5Key = "md5"
+	
+	func toJSONObject() -> Any {
+		return [
+			CachedFramework.nameKey: name,
+			CachedFramework.md5Key: md5
+		]
+	}
+}
+
+extension CachedFramework: Decodable {
+	static func decode(_ j: JSON) -> Decoded<CachedFramework> {
+		return curry(self.init)
+			<^> j <| CachedFramework.nameKey
+			<*> j <| CachedFramework.md5Key
+	}
+}
+
+private struct VersionFile {
+	let commitish: String
+	let xcodeVersion: String
+	
+	let macOS: [CachedFramework]?
+	let iOS: [CachedFramework]?
+	let watchOS: [CachedFramework]?
+	let tvOS: [CachedFramework]?
+	
+	static let commitishKey = "commitish"
+	static let xcodeVersionKey = "xcodeVersion"
+	
+	func cachesForPlatform(_ platform: Platform) -> [CachedFramework]? {
+		switch platform {
+		case .macOS:
+			return macOS
+		case .iOS:
+			return iOS
+		case .watchOS:
+			return watchOS
+		case .tvOS:
+			return tvOS
+		}
+	}
+	
+	func cachedPlatforms() -> Set<Platform> {
```

Comment:
Turns out this wasn't needed at all. The only call site was actually a bug. We need to use the set of all platforms rather than cachedPlatforms when provided with an empty platform set (basically no `--platform` option specified), so it's going to be removed.